days/src/day3/Day3.kt

```diff
@@ -0,0 +1,40 @@
+package day3
+
+import Day
+import java.io.File
+
+class Day3(private val terrain: List<CharArray>): Day {
+    data class Slope(val down: Int, val right: Int)
+    data class SlopeCount(val slope: Slope, var col: Int = 0, var trees: Int = 0) {
+        fun addTree(row: Int, terrain: CharArray) = this.also {
+            if (row % slope.down == 0 && terrain[col] == '#') { trees += 1 }
+        }
+        fun move(row: Int, max: Int) = this.also{
+            if (row % slope.down == 0) { col = (col + slope.right) % max }
+        }
+    }
+
+    private fun List<SlopeCount>.simulate(startCol: Int = 0) = this.also {
+        this.forEach { it.trees = 0; it.col = startCol }
+        terrain.forEachIndexed { row, terrain ->
```

Comment:
How does this work when the down slope is 2?

Reply:
That's handled in the move. It doesn't move the column or check the tree unless it's at the right level.

Reply:
It doesn't move or look at the trees unless it's on the right level.